linkerd/trace-context/src/service.rs

```diff
@@ -217,38 +221,33 @@ where
                 break 'outer;
             }
 
-            let span = self
-                .tracer
-                .span_builder("".to_string())
-                .start_with_context(&self.tracer, &parent_ctx);
+            let tracer = opentelemetry::global::tracer("");
```

Comment:
Did you consider caching this tracer in the context struct? I assume doing that would be less flexible, particularly if changing the tracer for testing.